app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(item) for item in errors],
+        "path": file_path,
+        "status": ("passed" if len(errors) == 0 else "failed"),
```

Comment:
One small note: In python you can use just "if errors" or "if not errors" to check the list length. "if errors" returns True if there are some elements inside this list, otherwise False
Also brackets are redundant here as there is no "and" "or" operators

Reply:
added needed changes